python/ngen_cal/src/ngen/cal/__main__.py

```diff
@@ -42,13 +43,21 @@ def main(general: General, model_conf: Mapping[str, Any]):
         import numpy as np
         np.random.seed(general.random_seed)
 
+    # model scope plugins setup in constructor
+    model = Model(model=model_conf)
+
+    # NOTE: if support for new models is added, this will need to be modified
+    assert isinstance(model.model, Ngen), f"ngen.cal.ngen.Ngen expected, got {type(model.model)}"
+    model_inner = model.model.unwrap()
+
     plugins = cast(List[Union[Callable, ModuleType]], general.plugins)
     plugin_manager = setup_plugin_manager(plugins)
 
     print(_loaded_plugins(plugin_manager))
 
     # setup plugins
     plugin_manager.hook.ngen_cal_configure(config=general)
+    model_inner._plugin_manager.hook.ngen_cal_model_configure(config=model_inner)
```

Comment:
An agent effectively side effects the `Model` it associates with, particularly setting the `workdir` for that model run.  Because of this, we probably need to initialize the Agent before passing the Model to anything else, including plugins.
```suggestion
    # Initialize the starting agent
    # WARNING: Agent sets the model workdir, so initialize it before using the model object for anything that may require an accurate workdir
    agent = Agent(model, general.workdir, general.log, general.restart, general.strategy.parameters)
    model_inner._plugin_manager.hook.ngen_cal_model_configure(config=model_inner)
```